### PR TITLE
Aztec Modal Announcement UI

### PR DESCRIPTION
**Pull request by frosty**
This PR adds the Aztec announcement UI to WPiOS.

aztec-announce

I've tried to make the alert relatively reusable, so that we can use it for upcoming login alerts. We'll need to make some tweaks there but I think it should be relatively straightforward. Essentially, the UI is configured by passing in a "configuration" object, which contains all of the label and button values / handlers. The view controller will update all its labels and resize as necessary.

A custom presentation is used to add a translucent view behind the modal.

To Test:

* Build and run the app (test with a device / sim with an existing build, and with a fresh install). When you first access a blog detail view controller, the alert should be displayed. Ensure it doesn't re-appear!
* Try dismissing the first modal screen.
* Try accepting the first modal screen and dismissing the second. The Aztec editor should be presented. Check also that App Settings shows that Aztec is enabled.

Notes:

* The blue links in the modals currently don't do anything, as we don't yet have an announcement post to link to.
* In a vertically compact layout (basically just 6+ / 7+ in landscape), the modal will hide its image view. This works okay with the modal content we have now, but ultimately we should probably put the whole thing in a scroll view in case it gets too long.

Needs review: @nheagy 
cc @iamthomasbishop

### CODE REVIEW

File: WordPress/Classes/ViewRelated/System/FancyAlertViewController.swift [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    override func accessibilityPerformEscape() -> Bool {
+        dismissTapped()
+        return true
+    }
+}
+
+private extension UIView {
+    /// Required to work around a bug in UIStackView where items don't become
+    /// hidden / unhidden correctly if you set their `isHidden` property
+    /// to the same value twice in a row. See http://www.openradar.me/22819594
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/System/FancyAlertViewController.swift]: What a ridiculous bug! Clean workaround tho 👍

**Comment by frosty** [comment]
Thanks for the comments, @nheagy! 😊

I was just resolving the merge conflict in the project file, when I thought of a cool extra feature... care to check it out? Now, when the alert has no buttons, you can fling it away to dismiss 😁

**Comment by nheagy** [comment]
Flinging the modal feels great, @frosty 

:shipit:

**Comment by frosty** [comment]
Thanks @nheagy!!

**Pull request merged**